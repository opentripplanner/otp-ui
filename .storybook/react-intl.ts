import flatten from "flat";

export const isTestRunner = () => window.navigator.userAgent.match(/StorybookTestRunner/);

/** Locale supported in the storybook "Globe" dropdown menu. */
const locales = ["en-US", "fr", "es", "vi", "ko", "zh", "unknown"];

/**
 * List of packages that will have localization support in Storybook.
 * FIXME: remove in favor of a loop on package names.
 */ 
const packages = [
  "base-map",
  "endpoints-overlay",
  "from-to-location-picker",
  "itinerary-body",
  "location-field",
  "printable-itinerary",
  "map-popup",
  "transit-vehicle-overlay",
  "trip-details",
  "trip-form"
];

/** Messages for all packages AND locales above. */
const messages = {};

// Populate messages if not running snapshots.
// (Message printouts would be unnecessary replicated in snapshots without that check.)
if (typeof window !== "undefined") {

  packages.forEach((pkg) => {
    locales.forEach((locale) => {
      // Chinese-simplified is assigned a special file name by Weblate.
      const localeFile = locale === "zh" ? "zh_Hans" : locale;
      try {
        messages[locale] = {
          ...messages[locale],
          ...flatten(require(`../packages/${pkg}/i18n/${localeFile}.yml`).default)
        };
      } catch (e) {
        // There is no yml files for the "unknown" locale,
        // so it should fail, and we won't display an error message in that case.
        if (locale !== "unknown") console.error(e);
      }    
    });
  });
}

// TODO: place any applicable (date, time, etc) format parameters here.
const formats = {};

export const reactIntl = {
  defaultLocale: "en-US",
  formats,
  locales,
  messages,
  timeZone: 'America/Los_Angeles'
};
