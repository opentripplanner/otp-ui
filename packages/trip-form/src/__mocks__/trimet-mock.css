@import url('https://fonts.googleapis.com/css?family=Hind&display=swap');
