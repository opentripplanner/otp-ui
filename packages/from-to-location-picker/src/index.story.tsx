import React, { ReactElement } from "react";
import { IntlProvider } from "react-intl";
import { action } from "@storybook/addon-actions";

import FromToLocationPicker from ".";

import "../__mocks__/trimet-mock.css";

const withIntl = (Story: () => JSX.Element) => (
  <IntlProvider messages={{}} locale="en">
    <Story />
  </IntlProvider>
);

export default {
  title: "From-To-Picker",
  component: FromToLocationPicker,
  decorators: [withIntl]
};

const onFromClick = action("onFromClick");
const onToClick = action("onToClick");

export const fromTo = (): ReactElement => (
  <FromToLocationPicker onFromClick={onFromClick} onToClick={onToClick} />
);

export const smallTextSansSerif = (): ReactElement => (
  <span style={{ fontSize: "75%", fontFamily: "sans-serif" }}>
    <FromToLocationPicker
      label
      onFromClick={onFromClick}
      onToClick={onToClick}
    />
  </span>
);

export const customStyleAndText = (): React.ReactElement => (
  <div className="trimet-ambient">
    <FromToLocationPicker
      label={<i style={{ color: "#ffff00" }}>Your trip:</i>}
      onFromClick={onFromClick}
      onToClick={onToClick}
      showIcons={false}
    />
  </div>
);
