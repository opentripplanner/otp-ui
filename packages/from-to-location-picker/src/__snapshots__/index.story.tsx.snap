// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From-To-Picker customStyleAndText smoke-test 1`] = `
<div class="trimet-ambient">
  <i style="color: rgb(255, 255, 0);">
    Your trip:
  </i>
  <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
    <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
      <button class="styled__Button-sc-vb4790-2 hNNoVB">
        Start here
      </button>
    </span>
    <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
      <button class="styled__Button-sc-vb4790-2 hNNoVB">
        Go here
      </button>
    </span>
  </span>
</div>
`;

exports[`From-To-Picker fromTo smoke-test 1`] = `
<span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
  <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
    <svg viewbox="0 0 512 512"
         height="0.9em"
         width="0.9em"
         aria-hidden="true"
         focusable="false"
         fill="currentColor"
         xmlns="http://www.w3.org/2000/svg"
         class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
    >
      <path fill="currentColor"
            d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
      >
      </path>
    </svg>
    <button class="styled__Button-sc-vb4790-2 hNNoVB">
      From here
    </button>
  </span>
  <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
    <svg viewbox="0 0 384 512"
         height="0.9em"
         width="0.9em"
         aria-hidden="true"
         focusable="false"
         fill="currentColor"
         xmlns="http://www.w3.org/2000/svg"
         class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
    >
      <path fill="currentColor"
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
      >
      </path>
    </svg>
    <button class="styled__Button-sc-vb4790-2 hNNoVB">
      To here
    </button>
  </span>
</span>
`;

exports[`From-To-Picker smallTextSansSerif smoke-test 1`] = `
<span style="font-size: 75%; font-family: sans-serif;">
  <strong>
    Plan a trip:
  </strong>
  <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
    <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
      <svg viewbox="0 0 512 512"
           height="0.9em"
           width="0.9em"
           aria-hidden="true"
           focusable="false"
           fill="currentColor"
           xmlns="http://www.w3.org/2000/svg"
           class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
      >
        <path fill="currentColor"
              d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
        >
        </path>
      </svg>
      <button class="styled__Button-sc-vb4790-2 hNNoVB">
        From here
      </button>
    </span>
    <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
      <svg viewbox="0 0 384 512"
           height="0.9em"
           width="0.9em"
           aria-hidden="true"
           focusable="false"
           fill="currentColor"
           xmlns="http://www.w3.org/2000/svg"
           class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
      >
        <path fill="currentColor"
              d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
        >
        </path>
      </svg>
      <button class="styled__Button-sc-vb4790-2 hNNoVB">
        To here
      </button>
    </span>
  </span>
</span>
`;
